Dockerfile.dapper

```diff
@@ -15,17 +15,17 @@ RUN if [ "${ARCH}" == "amd64" ]; then \
         curl -L "https://github.com/docker/compose/releases/download/1.29.2/docker-compose-Linux-x86_64" -o /usr/local/bin/docker-compose && \
         chmod +x /usr/local/bin/docker-compose; \
     fi
-RUN go install github.com/mgechev/revive@v1.1.1 && \
+RUN go install github.com/mgechev/revive@v1.2.4 && \
     rm -rf /go/src /go/pkg
 RUN go install golang.org/x/tools/cmd/goimports@latest && \
     rm -rf /go/src /go/pkg
 RUN if [ "$(go env GOARCH)" = "amd64" ]; then \
-        curl -sL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s v1.41.1; \
+        curl -sL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s v1.50.1; \
     fi
 RUN if [ "${ARCH}" == "amd64" ]; then \
-        go get sigs.k8s.io/kustomize/kustomize/v3@v3.5.4; \
+        go install sigs.k8s.io/kustomize/kustomize/v4@v4.5.7; \
```

Comment:
The reason that I pinned to v3.5.4 was because the kustomize folks broke something that I was leveraging in v3.5.5 (cannot remember what). That in itself is a reasonable enough argument to keep on pinning but that is warring with my sense that I like to get all the goodies that can some with updates on the branch stream.

Hmm 🤔 .

Well, I no longer work at Rancher/SUSE, but thems my thoughts.

Reply:
@dweomer thanks for adding context to the issue. I'll close this PR giving that Brandon is working on https://github.com/rancher/system-upgrade-controller/pull/240.